### PR TITLE
[4.12 4.13] Apply sts:AssumeRole to ingress operator policy for AWS Shared-VPC cluster

### PR DESCRIPTION
**Pull request by yunjiang29**
Address https://issues.redhat.com//browse/OCPBUGS-17733

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-aws-ipi-disconnected-sts-p2-f28

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-proxy-sts-p1-f28-destructive

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-proxy-sts-p1-f28-destructive

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-proxy-sts-p1-f28-destructive

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-proxy-sts-p1-f28-destructive

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-upgrade-from-stable-4.13-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-shared-vpc-phz-sts-fips-f14

**Comment by yunjiang29** [comment]
## test 

```
```

**Comment by yunjiang29** [comment]
@jianlinliu PTAL, thanks

File: ci-operator/step-registry/ipi/conf/aws/oidc-creds-provision/ipi-conf-aws-oidc-creds-provision-commands.sh [hunk 1]
```diff
@@ -22,6 +22,28 @@ echo "RELEASE_IMAGE_LATEST_FROM_BUILD_FARM: ${RELEASE_IMAGE_LATEST_FROM_BUILD_FA
 oc registry login
 oc adm release extract --credentials-requests --cloud=aws --to="/tmp/credrequests" "${RELEASE_IMAGE_LATEST_FROM_BUILD_FARM}"
 
+if [[ ${ENABLE_SHARED_VPC} == "yes" ]]; then
+  ingress_cr_file="/tmp/credrequests/0000_50_cluster-ingress-operator_00-ingress-credentials-request.yaml"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/oidc-creds-provision/ipi-conf-aws-oidc-creds-provision-commands.sh:26]: Shall we add one more check the file exists or not?

Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/oidc-creds-provision/ipi-conf-aws-oidc-creds-provision-commands.sh]: Updated

**Pull request merged**